ssz-rs-derive/src/lib.rs

```diff
@@ -392,44 +395,27 @@ fn validate_derive_data(data: &Data) {
     }
 }
 
-#[proc_macro_derive(SimpleSerialize)]
-pub fn derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
-    let input = parse_macro_input!(input as DeriveInput);
-
-    let data = &input.data;
-    validate_derive_data(data);
-
-    let name = &input.ident;
-    let generics = &input.generics;
+fn impl_serilizable(
+    data: &Data,
+    name: &Ident,
+    impl_generics: &ImplGenerics,
+    ty_generics: &TypeGenerics,
+) -> proc_macro2::TokenStream {
     let serialize_impl = derive_serialize_impl(data);
     let deserialize_impl = derive_deserialize_impl(data);
     let is_variable_size_impl = derive_variable_size_impl(data);
     let size_hint_impl = derive_size_hint_impl(data);
-    let merkleization_impl = derive_merkleization_impl(data);
 
-    let impl_impl = if generics.params.is_empty() {
-        quote! { impl }
-    } else {
-        quote! { impl #generics }
-    };
-
-    let name_impl = if generics.params.is_empty() {
```

Comment:
Actually, this condition would not be necessary.  `split_for_impl` would do the trick and leave blank if type generic is empty on `quote!`.